src/snowflake/cli/api/project/schemas/native_app/path_mapping.py

```diff
@@ -1,10 +1,43 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import List, Literal, Optional
 
+from pydantic import Field
 from snowflake.cli.api.project.schemas.updatable_model import UpdatableModel
 
+# Limited to python options only for now.
+ProcessorOptions = Literal["python-snowpark", "PYTHON-SNOWPARK"]
```

Comment:
I'd use snowpark-python here, or even just snowpark. We can infer it's python (vs java later on).

Also, the plugin infrastructure will require this to be a string. I'd bake that in upfront and just have dynamic validation of the processor name with the one supported value later in the pipeline.